lib/ansible/modules/cloud/kubevirt/kubevirt_pvc.py

```diff
@@ -0,0 +1,399 @@
+    cdi_source:
+      blank: yes
+
+- name: Create a PVC and fill it with data from a container
+  kubevirt_pvc:
+    name: pvc5
+    namespace: default
+    size: 100Mi
+    access_modes:
+      - ReadWriteOnce
+    cdi_source:
+      registry:
+        url: "docker://kubevirt/fedora-cloud-registry-disk-demo"
+
+'''
+
+RETURN = '''
+result:
+  description:
+  - The created, patched, or otherwise present object. Will be empty in the case of a deletion.
+  returned: success
+  type: complex
+  contains:
+     api_version:
+       description: The versioned schema of this representation of an object.
+       returned: success
+       type: str
+     kind:
+       description: Represents the REST resource this object represents.
+       returned: success
+       type: str
+     metadata:
+       description: Standard object metadata. Includes name, namespace, annotations, labels, etc.
+       returned: success
+       type: complex
+     spec:
+       description: Specific attributes of the object. Will vary based on the I(api_version) and I(kind).
+       returned: success
+       type: complex
+     status:
+       description: Current status details for the object.
+       returned: success
+       type: complex
+     items:
+       description: Returned only when multiple yaml documents are passed to src or resource_definition
+       returned: when resource_definition or src contains list of objects
+       type: list
+     duration:
+       description: elapsed time of task in seconds
+       returned: when C(wait) is true
+       type: int
+       sample: 48
+'''
+
+
+import copy
+import traceback
+
+from collections import defaultdict
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC
+from ansible.module_utils.k8s.raw import KubernetesRawModule
+from ansible.module_utils.kubevirt import virtdict, KubeVirtRawModule
+
+
+PVC_ARG_SPEC = {
+    'name': {'required': True},
+    'namespace': {'required': True},
+    'state': {
+        'type': 'str',
+        'choices': [
+            'present', 'absent'
+        ],
+        'default': 'present'
+    },
+    'force': {
+        'type': 'bool',
+        'default': False,
+    },
+    'merge_type': {
+        'type': 'list',
+        'choices': ['json', 'merge', 'strategic-merge']
+    },
+    'resource_definition': {
+        'type': 'dict',
+        'aliases': ['definition', 'inline']
+    },
+    'labels': {'type': 'dict'},
+    'annotations': {'type': 'dict'},
+    'selector': {'type': 'dict'},
+    'access_modes': {'type': 'list'},
+    'size': {
+        'type': 'str',
+        'aliases': ['storage']
+    },
+    'storage_class_name': {'type': 'str'},
+    'volume_mode': {'type': 'str'},
+    'volume_name': {'type': 'str'},
+    'cdi_source': {'type': 'dict'}
+}
+
+
+class KubevirtPVC(KubernetesRawModule):
+    def __init__(self):
+        super(KubevirtPVC, self).__init__(
+            required_if=[
+                ('state', 'present', ['size', 'access_modes']),
```

Comment:
Would be nice to have this requirment written in the documentation. Note that with this requirment you will have to specify `size` parameter always even when user wants to just patch `access_modes` for example.

Reply:
Dropped.